cf/commands/servicekey/delete_service_key.go

```diff
@@ -86,7 +86,7 @@ func (cmd DeleteServiceKey) Run(c *cli.Context) {
 	}
 
 	serviceKey, err := cmd.serviceKeyRepo.GetServiceKey(serviceInstance.Guid, serviceKeyName)
-	if err != nil {
+	if err != nil || serviceKey.Fields.Guid == "" {
```

Comment:
the Guid could be empty if the service key doesn't  exist but empty result is returned without an error by director. The error need also be checked.